airbyte_cdk/sources/declarative/retrievers/async_retriever.py

```diff
@@ -74,14 +78,36 @@ def _validate_and_get_stream_slice_jobs(
         """
         return stream_slice.extra_fields.get("jobs", []) if stream_slice else []
 
+    def _get_cursor_slice_info(
+        self,
+        stream_slice: Optional[StreamSlice] = None,
+    ) -> Mapping[str, Any]:
+        """
+        Retrieve the cursor slice information from a provided stream slice.
+
+        This method checks if a stream slice is provided and contains a 'cursor_slice' attribute.
+        If present, it returns the value of 'cursor_slice' as a dictionary. Otherwise, it returns an empty dictionary.
+
+        Args:
+            stream_slice (Optional[StreamSlice]): An optional stream slice object that may include a 'cursor_slice' attribute.
+
+        Returns:
+            Mapping[str, Any]: A dictionary containing the cursor slice information if available, otherwise an empty dictionary.
+        """
+        return stream_slice.cursor_slice if stream_slice and stream_slice.cursor_slice else {}
```

Comment:
`partition` and `cursor_slice` are retired fields used to initiate `stream_slice`. So, I don't think there will be a case where `cursor_slice` does not exist.

Reply:
I've removed the `_get_slice_cursor()` method and call the `stream_slice.cursor_slice` directly now with ignoring for possible `None` case.